provider/maas/instance.go

```diff
@@ -262,3 +263,21 @@ func (mi *maas1Instance) hostname() (string, error) {
 	// A MAAS instance has its DNS name immediately.
 	return mi.maasObject.GetField("hostname")
 }
+
+func (mi *maas1Instance) fqdn() (string, error) {
+	return mi.maasObject.GetField("fqdn")
+}
+
+func (mi *maas1Instance) displayName() (string, error) {
+	displayName, err := mi.hostname()
+	if err != nil {
+		logger.Infof("error detecting hostname for %s", mi)
+	}
+	if displayName == "" {
+		displayName, err = mi.fqdn()
```

Comment:
Isn't fqdn just fmt.Sprintf("%s.%s", hostname, domainname) ?

how is it that we we would have an FQDN but not a hostname?

Reply:
Older versions of MAAS provided fqdn but not hostname. Like 2.3 - which much of bootstack is using.